synapse/storage/engines/postgres.py

```diff
@@ -65,7 +70,7 @@ def check_database(self, db_conn, allow_outdated_version: bool = False):
         # docs: The number is formed by converting the major, minor, and
         # revision numbers into two-decimal-digit numbers and appending them
         # together. For example, version 8.1.5 will be returned as 80105
-        self._version = db_conn.server_version
+        self._version = cast(int, db_conn.server_version)
```

Comment:
If accepted, https://github.com/python/typeshed/pull/7834 would make this cast redundant.